File: 01_projects/code/java-playwright/cbaps-dawms/src/test/java/com/dawms/pages/SignatureRoutingPage.java
package com.dawms.pages;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import com.microsoft.playwright.Locator;
import com.microsoft.playwright.Page;
import com.dawms.library.PlaywrightManager;

/**
 * SignatureRoutingPage: Route submission for signature approval
 * Select signer and submit for signature
 */
public class SignatureRoutingPage {

	private static final Logger log = LoggerFactory.getLogger(SignatureRoutingPage.class);
	private final Page page;
	private final PlaywrightManager pwm;

	private final Locator signerDropdown;
	private final Locator routeForSignatureButton;

	public SignatureRoutingPage(Page page, PlaywrightManager pwm) {
		this.page = page;
		this.pwm = pwm;

		// Initialize locators
		this.signerDropdown = page.locator("#signer");
		this.routeForSignatureButton = page.locator("button:has-text('Submit for Signature')");

		// Stability anchor: wait for signature routing page to load
		pwm.waitVisible("text=Signature Routing");
		log.info("Signature Routing Page loaded successfully.");
	}

	/**
	 * Submit submission for signature to selected signer
	 * @param signerRoleOrName - Signer role or name
	 * @return MilestoneStatusPage
	 */
	public MilestoneStatusPage submitForSignature(String signerRoleOrName) {
		signerDropdown.selectOption(signerRoleOrName);
		log.info("Selected signer: " + signerRoleOrName);
		
		pwm.click(routeForSignatureButton);
		log.info("Clicked 'Submit for Signature' button.");
		
		return new MilestoneStatusPage(page, pwm);
	}
}

File: 01_projects/code/python-playwright/cbaps-dawms/basic-pom/dawms/pages/signature_routing_page.py
"""
SignatureRoutingPage: Route submission for signature approval
Select signer and submit for signature
"""

import logging
from playwright.sync_api import Page, Locator
from dawms.library.playwright_manager import PlaywrightManager

logger = logging.getLogger(__name__)


class SignatureRoutingPage:
    """
    SignatureRoutingPage handles signature routing workflow.
    This is the Python equivalent of the Java SignatureRoutingPage.
    """
    
    def __init__(self, page: Page, pwm: PlaywrightManager):
        """
        Initialize SignatureRoutingPage
        
        Args:
            page: Playwright Page object
            pwm: PlaywrightManager instance
        """
        self.page = page
        self.pwm = pwm
        
        # Locators
        self.signer_dropdown: Locator = page.locator("#signer")
        self.route_for_signature_button: Locator = page.locator(
            "button:has-text('Submit for Signature')"
        )
        
        # Stability anchor: wait for signature routing page to load
        pwm.wait_visible("text=Signature Routing")
        logger.info("Signature Routing Page loaded successfully.")
    
    def submit_for_signature(self, signer_role_or_name: str) -> 'MilestoneStatusPage':
        """
        Submit submission for signature to selected signer
        
        Args:
            signer_role_or_name: Signer role or name
            
        Returns:
            MilestoneStatusPage instance
        """
        from dawms.pages.milestone_status_page import MilestoneStatusPage
        
        self.signer_dropdown.select_option(signer_role_or_name)
        logger.info(f"Selected signer: {signer_role_or_name}")
        
        self.pwm.click(self.route_for_signature_button)
        logger.info("Clicked 'Submit for Signature' button.")
        
        return MilestoneStatusPage(self.page, self.pwm)
